### ISSUE TITLE
Disable LeavesEstimationIterations check

### ISSUE THREAD

**Issue by david-waterworth** [catboost/catboost#723]
This is a potential quick fix for https://github.com/catboost/catboost/issues/719 

By removing the restriction `gradient_iterations should equals 1 for this mode` I get much better convergence when using ELossFunction::Quantile

It appears to work but I've only tested CPU training.

catboost_options.cpp

    ELeavesEstimation leavesEstimation = ObliviousTreeOptions->LeavesEstimationMethod;
    if (lossFunction == ELossFunction::Quantile ||
        lossFunction == ELossFunction::MAE ||
        lossFunction == ELossFunction::LogLinQuantile ||
        lossFunction == ELossFunction::MAPE)
    {
        CB_ENSURE(leavesEstimation != ELeavesEstimation::Newton,
                  "Newton leave estimation method is not supported for " << lossFunction << " loss function");
        //CB_ENSURE(ObliviousTreeOptions->LeavesEstimationIterations == 1U,
        //          "gradient_iterations should equals 1 for this mode");
    }

I also have a feeling catboost should start training from a better initial approximation, i.e. the mean / median or quantile depending on the loss function, rather than zero. Otherwise convergence is overly impacted by the distribution of the training data?

**Comment by david-waterworth** [comment]
I tested on GPU and there doesn't appear to be any issues with `--leaf-estimate-iterations` > 1 for `Quantile:alpha=0.75` loss

I ran a quick experiment. Setting learning rate to 1, to converge to the exact minima of the loss function (0.925) requires  --leaf-estimate-iterations * --iterations ~ 100. Using GPU, it's much more efficient to do 100 gradient iterations on a single tree than 1 gradient iteration on 100 trees (by a factor of about 5).

--leaf-estimate-iterations | --iterations | time
1 | 100 | 54.3ms
10 | 10 | 15.8ms
100 | 1 | 8.28ms

`catboost fit --task-type GPU --leaf-estimation-iterations 100 --iterations 1 --learning-rate 1 --depth 1 --loss-function Quantile:alpha=0.75 --learn-set ${workspaceFolder}/catboost/app/data/train_small --column-description ${workspaceFolder}/catboost/app/data/train.cd --boosting-type Plain`

train.cd
> 0	Num
> 1	Target

train_small
> 1.0	1.0
> 1.0	2.0
> 1.0	3.0
> 1.0	4.0
> 1.0	5.0
> 1.0	6.0
> 1.0	7.0
> 1.0	8.0
> 1.0	9.0
> 1.0	10.0
> 2.0	21.0
> 2.0	22.0
> 2.0	23.0
> 2.0	24.0
> 2.0	25.0
> 2.0	26.0
> 2.0	27.0
> 2.0	28.0
> 2.0	29.0
> 2.0	30.0

**Comment by david-waterworth** [comment]
And a final experiment, starting the boosting from a non-zero baseline

Freidman sets the baseline to the values which minimise the loss function over the full dataset - which implies using the quantile of y. But in my test case this didn't produce any improvement. Using the mean did though, it reduced the iterations to convergence from 100 to 75

```
from catboost import Pool
train_data = Pool(X, y)
a0 = np.mean(y)
#a0 = np.quantile(y,alpha)
train_data.set_baseline(np.repeat(a0,20))

model = CatBoostRegressor(task_type='GPU',learning_rate=1,leaf_estimation_iterations=1,iterations=100,depth=1,loss_function=f'Quantile:alpha=0.75',boosting_type='Plain')
model.fit(train_data);

model.predict(X)+a0
```
